0000-macro-value.md

```diff
@@ -0,0 +1,300 @@
+- Feature Name: `macro_value`
+- Start Date: 2021-03-27
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Add a new macro_rules matcher, `$name:value`, with identical semantics to that of a function capture.
+
+# Motivation
+[motivation]: #motivation
+
+Value arguments to function-like macros are tricky to deal with.
+While macro_rules macros don't suffer from the common and most egregious pitfalls of C-style preprocessor macros,
+such as misnested brackets and operator precedence,
```

Comment:
Hmm... Are you sure this "jab" contributes to the RFC?

Reply:
I replaced

> the common and most egregious pitfalls of C-style

with just

> the most common pitfalls of C-style

to remove the blame connotation.

An explicit design choice for `macro_rules!` was hygenic macros that avoid the "dumb token substitution" problems that result from simple preprocessor macros as provided by C. I do think it's worth connecting this feature to that pattern of removing misexpansion from naive macros.